Null hypothesis:new bus has equal or more waiting time than the old bus line.
alpha = 0.05
Ho:T(new)>=T(old)

In [9]:
#written by YuchenDing
import pandas as pd 
import numpy as np 
import json
import matplotlib.pyplot as plt 
import seaborn as sns

In [11]:
#rawdata
!wget https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt

--2018-10-03 22:01:46--  https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.200.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.200.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1179 (1.2K) [text/plain]
Saving to: ‘times.txt.2’

times.txt.2         100%[===================>]   1.15K  --.-KB/s    in 0s      

2018-10-03 22:01:47 (234 MB/s) - ‘times.txt.2’ saved [1179/1179]



In [12]:
rawdata = pd.read_csv("times.txt", header=None)

In [14]:
rawdata.describe()
rawdata.head(8)

0
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550
5  34.822079
6  39.871884
7  39.579946

In [13]:
Mu, TIME, LEN, SIGMA = 36.0, float(rawdata.mean()), len(rawdata), float(rawdata.std())
calc_z = lambda time, mu, sigma, n : (time - mu) / (sigma / (np.sqrt(n)))
z_score = calc_z(TIME, Mu, SIGMA, LEN)
print("Calculated z-score: %f Abs z-score: %f" % (z_score, abs(z_score)))

Calculated z-score: -2.149052 Abs z-score: 2.149052


z-score in this case is 2.149052, which is greater than critical value(1.64), at significance level 0.05.
We should reject the null hypothesis,so the new bus waiting time is less than the old bus line.